shell/platform/linux/public/flutter_linux/fl_basic_message_channel.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_BASIC_MESSAGE_CHANNEL_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_BASIC_MESSAGE_CHANNEL_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <gio/gio.h>
+#include <glib-object.h>
+
+#include "fl_binary_messenger.h"
+#include "fl_message_codec.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlBasicMessageChannel,
+                         fl_basic_message_channel,
+                         FL,
+                         BASIC_MESSAGE_CHANNEL,
+                         GObject)
+
+/**
+ * FlBasicMessageChannel:
+ *
+ * #FlBasicMessageChannel is an object that allows sending and receiving
+ * messages to/from Dart code over platform channels.
+ *
+ * #FlBasicMessageChannel matches the BasicMessageChannel class in the Flutter
+ * services library.
+ */
+
+struct _FlBasicMessageChannelClass {
+  GObjectClass parent_class;
+};
+
+/**
+ * FlBasicMessageChannelMessageHandler:
+ * @channel: a #FlBasicMessageChannel
+ * @message: message received
+ * @user_data: (closure): data provided when registering this handler
+ *
+ * Function called when a message is received.
+ */
+typedef void (*FlBasicMessageChannelMessageHandler)(
+    FlBasicMessageChannel* channel,
+    FlValue* message,
+    gpointer user_data);
+
+/**
+ * fl_basic_message_channel_new:
+ * @messenger: a #FlBinaryMessenger
+ * @name: a channel name
+ * @codec: the message codec
+ *
+ * Create a new basic message channel. @codec must match the codec used on the
+ * Dart end of the channel.
+ *
+ * Returns: a new #FlBasicMessageChannel.
+ */
+FlBasicMessageChannel* fl_basic_message_channel_new(
+    FlBinaryMessenger* messenger,
+    const gchar* name,
+    FlMessageCodec* codec);
+
+/**
+ * fl_basic_message_channel_set_message_handler:
+ * @channel: a #FlBasicMessageChannel
+ * @handler: (allow-none): function to call when a message is received on this
+ * channel or %NULL to disable the handler.
+ * @user_data: (closure): user data to pass to @handler
+ *
+ * Set the function called when a message is received.
+ */
+void fl_basic_message_channel_set_message_handler(
+    FlBasicMessageChannel* channel,
+    FlBasicMessageChannelMessageHandler handler,
+    gpointer user_data);
+
+/**
+ * fl_basic_message_channel_send:
+ * @channel: a #FlBasicMessageChannel
+ * @message: message to send, must match what the #FlMessageCodec supports
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL
+ *
+ * Send a message.
+ *
+ * Returns: %TRUE on success
+ */
+gboolean fl_basic_message_channel_send(FlBasicMessageChannel* channel,
```

Comment:
There should be an optional way to get a reply here; I'm fine with this landing as-is, but you should add a TODO for the reply.

Reply:
Oh, I misread what BasicMessageChannel does, I thought these were events, not request/replies. I'll update the API to do that.